### PR TITLE
Only strip VCS prefix for URIs with vcs+ patterns

### LINKED ISSUE #1126: Pipenv fails to install from github by ref and tag
Trying to install from github using ref and tag fails, because pipenv "destroys" the format of the url removing `git:` from the reference.

##### Describe your environment

1. OS Type OSX 10.12.6
1. Python version: `$ python -V` 3.6.3
1. Pipenv version: `$ pipenv --version` 8.3.2

##### Expected result

The reference in `Pipfile` looks like this (which is correct)
`pytest-tags = {git = "git://github.com/Projectplace/pytest-tags.git", editable = true, ref = "1.5"}`

But in the `Pipfile.lock` file, it's
`"git": "//github.com/Projectplace/pytest-tags.git"`

Workaround is to re-add `git:` to the pipfile.lock manually.

##### Actual result

```Pipfile.lock not found, creating…
Locking [dev-packages] dependencies…
Locking [packages] dependencies…
Updated Pipfile.lock (24ab2e)!
Installing dependencies from Pipfile.lock (24ab2e)…
  🐍   ▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉ 19/19 — 00:00:05
An error occurred while installing -e git+//github.com/Projectplace/pytest-tags.git@c8cf057c03a1fa30769c0328efa41d4826e03bcf#egg=pytest-tags! Will try again.
Installing initially–failed dependencies…
Obtaining pytest-tags from git+//github.com/Projectplace/pytest-tags.git@c8cf057c03a1fa30769c0328efa41d4826e03bcf#egg=pytest-tags
  Cloning github.com/Projectplace/pytest-tags.git (to c8cf057c03a1fa30769c0328efa41d4826e03bcf) to /Users/jimbrannlund/.local/share/virtualenvs/dpo-automation-NuB7OARD/src/pytest-tags
fatal: repository 'github.com/Projectplace/pytest-tags.git' does not exist

Command "git clone -q github.com/Projectplace/pytest-tags.git /Users/jimbrannlund/.local/share/virtualenvs/dpo-automation-NuB7OARD/src/pytest-tags" failed with error code 128 in None```

### PR DESCRIPTION
**Pull request by techalchemy**
- Fixes #1126
- Does a simple check before stripping the VCS prefix on
a `requirement.uri` in `pipenv.utils.convert_deps_from_pip`

### CODE REVIEW

File: pipenv/utils.py [hunk 1]
```diff
@@ -45,7 +45,7 @@
 
 # List of version control systems we support.
 VCS_LIST = ('git', 'svn', 'hg', 'bzr')
-SCHEME_LIST = ('http://', 'https://', 'ftp://', 'file://')
```
Comment (nateprewitt) [inline pipenv/utils.py]: Why is this getting added to the scheme list?

File: pipenv/utils.py [hunk 2]
```diff
@@ -661,8 +661,10 @@ def convert_deps_from_pip(dep):
             req.path = None
 
         # Crop off the git+, etc part.
-        dependency.setdefault(req.name, {}).update({req.vcs: req.uri[len(req.vcs) + 1:]})
-
+        if '+' in req.uri:
```
Comment (nateprewitt) [inline pipenv/utils.py]: So I don't know if the `+` check by itself will be sufficient. That's a legal character in the path as well. We need to determine if there's a + explicitly in the scheme.

File: pipenv/utils.py [hunk 3]
```diff
@@ -661,8 +661,10 @@ def convert_deps_from_pip(dep):
...
+            req.uri = req.uri[len(req.vcs) + 1:]
+        dependency.setdefault(req.name, {}).update({req.vcs: req.uri})
+        print('dependency: {}'.format(dependency))
```
Comment (nateprewitt) [inline pipenv/utils.py]: Let's remove this.

Comment (techalchemy) [reply pipenv/utils.py]: ooh my bad

File: pipenv/utils.py (hunk 2 above)
Comment (techalchemy) [reply pipenv/utils.py]: yeah good call

File: pipenv/utils.py (hunk 1 above)
Comment (techalchemy) [reply pipenv/utils.py]: erm, it shouldn't be necessary, sorry for the sloppy PR that was my early hope that just adding it to the scheme list would pick it up in the logic

**Pull request merged**